src/main/java/ServerConnectionException.java

```diff
@@ -0,0 +1,3 @@
+public class ServerConnectionException extends RuntimeException {
```

Comment:
I believe the reasoning behind this was that by creating my own exception it was unchecked, so it would pass at runtime. IOException is a checked exception and would prevent the test from running?

Reply:
ah ok i see. alternatively you can `throw IOException` in your test files, but this works too.